src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -564,6 +580,27 @@ private TypeGenerationSpec ParseTypeGenerationSpec(in TypeToGenerate typeToGener
                 }
             }
 
+            private bool IsMemoryType(ITypeSymbol type, [NotNullWhen(true)] out INamedTypeSymbol? namedType, out CollectionType collectionType)
```

Comment:
Why not just inline this logic in the main collection resolution method?

Reply:
Becaose of https://github.com/dotnet/runtime/pull/88713#discussion_r1261279296. Since I couldn't rely on registering the types as built-in supported, doing everything in here avoids redundant checks.